### PR TITLE
Fav bookmark

### LINKED ISSUE #256: Mentee bookmark fav mentors
#### Description
Allow mentees to bookmark their favorite mentors!

#### Tasks
Include specific tasks in the order they need to be done in. Include links to specific lines of code where the task should happen at.
- [x] add field of bookmark_mentors to the mentee model. This should be a list of mentor's ids
- [x] add bookmark button in mentor profile on the gallery cards and their actual page
- [x] display their favorite mentors in the "Favorites" tab (as of mar 29 faith is still working on designs)

### PR DESCRIPTION
**Pull request by chenfeiyu132**
## Status:
🚀 : Ready

## Description :star2:

Fixes #256 

Added favoriteMentors Listfield for MenteeProfile in the backend, added button on front-end for mentee user to bookmark their favorite mentors.
Screen Shot 2021-04-15 at 8 13 28 PM

 
## TODOs :star:

- [ ] Write endpoint to update mentee profile with favorite mentors
- [ ] Connect endpoint with frontend functions

### CODE REVIEW

File: backend/api/models/Message.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from api.core import Mixin
```
Comment (Leundai) [inline backend/api/models/Message.py:1]: Is the code for messages supposed to be included with this PR too?

File: backend/api/views/mentee.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+def add_fav_mentor():
+    try:
+        data = request.get_json()
+        mentor_uid = data["mentor_uid"]
+        mentee_uid = data["mentee_uid"]
+    except:
+        msg = "invalid parameters provided"
+        logger.info(msg)
+        return create_response(status=422, message=msg)
+    print()
```
Comment (Leundai) [inline backend/api/views/mentee.py:21]: remove

File: frontend/src/components/MentorCard.js [hunk 3]
```diff
@@ -46,6 +53,12 @@ function MentorCard(props) {
     }
   }
 
+  function onFavoriteClick() {
+    setFavorite(!favorite);
+    console.log("hit");
```
Comment (Leundai) [inline frontend/src/components/MentorCard.js:58]: this too

File: frontend/src/components/pages/Gallery.js [hunk 4]
```diff
@@ -30,6 +35,37 @@ function Gallery() {
...
+    if (isMentee) {
+      getMentee();
+    }
+  }, [isMentee]);
+
+  useEffect(() => {
+    function initializeFavorites() {
+      let fav_set = new Set();
+      mentee.favorite_mentors_uids.forEach((uid) => {
+        console.log(uid);
```
Comment (Leundai) [inline frontend/src/components/pages/Gallery.js:55]: this as well

**Review by Leundai — APPROVED** [review]
Looking good! :)

**Review by AngelaLuo49021 — APPROVED** [review]
CUTEEEEEE

**Pull request merged**